### PR TITLE
Migrate export management cmd

### PR DESCRIPTION
**Pull request by benrudolph**
@dimagi/scale-team this is the command to migrate over all exports when we're ready 

cc: @sravfeyn

### CODE REVIEW

**Comment by benrudolph** [comment]
:fish: but probably small enough for 🏢

File: corehq/apps/reports/tests/test_dbaccessors.py [hunk 1]
```diff
@@ -1,8 +1,15 @@
+import json
```
Comment (dimagimon) [inline corehq/apps/reports/tests/test_dbaccessors.py]: F401 'json' imported but unused

File: corehq/apps/export/management/commands/migrate_exports.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
+from django.core.management.base import BaseCommand
+
+from couchexport.models import SavedExportSchema
```
Comment (dimagimon) [inline corehq/apps/export/management/commands/migrate_exports.py]: F401 'SavedExportSchema' imported but unused

File: corehq/apps/reports/dbaccessors.py [hunk 3]
```diff
@@ -28,12 +29,13 @@ def stale_get_exports_json(domain):
...
         include_docs=False,
-        limit=1
-    ).count()
+        limit=1,
+        reduce=True,
+    ).one()
+    return result["value"] if result else 0
 
 
 def touch_exports(domain):
```
Comment (benrudolph) [inline corehq/apps/reports/dbaccessors.py]: i don't know how any of these functions worked before

**Comment by benrudolph** [comment]
✅ tests on 2nd to last commit, lint review on last commit

File: corehq/apps/export/management/commands/migrate_exports.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        for doc in Domain.get_all(include_docs=False):
+            domain = doc['key']
+            export_count = stale_get_export_count(domain)
+            if export_count:
+                for old_export in with_progress_bar(
+                        stale_get_exports_json(domain),
+                        length=export_count,
+                        prefix=domain):
+                    try:
+                        convert_saved_export_to_export_instance(domain, old_export)
```
Comment (sravfeyn) [inline corehq/apps/export/management/commands/migrate_exports.py]: Is there no way to do this in 'bulk'?

File: corehq/util/log.py [hunk 5]
```diff
@@ -235,7 +235,7 @@ def filter(self, record):
             return False
 
 
-def with_progress_bar(iterable, length=None):
+def with_progress_bar(iterable, length=None, prefix='Processing'):
```
Comment (NoahCarnahan) [inline corehq/util/log.py]: cool, didn't know about this function.

File: corehq/apps/export/management/commands/migrate_exports.py (hunk 4 above)
Comment (benrudolph) [inline corehq/apps/export/management/commands/migrate_exports.py]: Could use gevent to spawn multiple of these tasks, but don't think it's worth it. I suspect there aren't that many exports and this is only going to be run once

File: corehq/util/log.py (hunk 5 above)
Comment (benrudolph) [inline corehq/util/log.py]: i believe this is @esoergel's baby. pretty nice little utility!

**Comment by benrudolph** [comment]
this does have a reindex that needs to be done on the main hq db. @dannyroberts do you think that's OK? or should we revert and wait til friday?

**Comment by NoahCarnahan** [comment]
Ah shit, missed the view change, sorry.

**Comment by benrudolph** [comment]
no worries, i should've added the purple label

File: corehq/util/log.py (hunk 5 above)
Comment (esoergel) [inline corehq/util/log.py]: :blush:

**Comment by esoergel** [comment]
Huh, I would've expected that view to be in `commcarehq__meta`, but I just checked and it's definitely in the primary db.

**Comment by benrudolph** [comment]
yeah, export data stuff is split between those two dbs, but everything couchexport is in the main db. the new exports ensure that everything is in `__meta`

**Pull request merged**